app/src/main/java/com/asfoundation/wallet/repository/BdsBuyService.java

```diff
@@ -0,0 +1,168 @@
+package com.asfoundation.wallet.repository;
+
+import android.support.annotation.NonNull;
+import com.asf.wallet.BuildConfig;
+import com.asfoundation.wallet.entity.TokenInfo;
+import com.asfoundation.wallet.entity.TransactionBuilder;
+import com.asfoundation.wallet.interact.DefaultTokenProvider;
+import com.asfoundation.wallet.poa.CountryCodeProvider;
+import com.asfoundation.wallet.poa.DataMapper;
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import java.math.BigDecimal;
+import java.util.List;
+
+/**
+ * Created by trinkes on 3/16/18.
+ */
+
+public class BdsBuyService implements BuyService {
```

Comment:
this is a generic buy service and not a bds one, do we really need an interface?